File: 3570-count-of-substrings-containing-every-vowel-and-k-consonants-i/3570-count-of-substrings-containing-every-vowel-and-k-consonants-i.cpp
class Solution {
public:
    int countOfSubstrings(string w,int k){
        unordered_set<char> mp= {'a', 'e', 'i', 'o', 'u'};
        int ans= 0;
        int n= w.length();

        for(int i=0; i<n; i++){
            unordered_set<char> f;
            int cnt= 0;
            for(int j=i; j<n; j++){
                if(mp.count(w[j])){
                    f.insert(w[j]);
                }
                else{
                    cnt++;
                }
                if(cnt>k){
                    break;
                }
                if(f.size()==5 && cnt== k){
                    ans++;
                }
            }
        }
    return ans;
    }

};

File: 3570-count-of-substrings-containing-every-vowel-and-k-consonants-i/3570-count-of-substrings-containing-every-vowel-and-k-consonants-i.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int countOfSubstrings(String w, int k) {
        Set<Character> vowels = Set.of('a', 'e', 'i', 'o', 'u');
        int ans = 0;
        int n = w.length();

        for (int i = 0; i < n; i++) {
            Set<Character> f = new HashSet<>();
            int vowelCount = 0;

            for (int j = i; j < n; j++) {
                if (vowels.contains(w.charAt(j))) {
                    f.add(w.charAt(j));
                    vowelCount++;
                }
                int nonVowelCount = (j - i + 1) - vowelCount;

                if (nonVowelCount > k) {
                    break;
                }
                if (f.size() == 5 && nonVowelCount == k) {
                    ans++;
                }
            }
        }
        return ans;
    }
}
